src/Juvix/Compiler/Core/Data/InfoTableBuilder.hs

```diff
@@ -12,8 +12,9 @@ data InfoTableBuilder m a where
   RegisterConstructor :: ConstructorInfo -> InfoTableBuilder m ()
   RegisterInductive :: InductiveInfo -> InfoTableBuilder m ()
   RegisterIdentNode :: Symbol -> Node -> InfoTableBuilder m ()
+  RegisterMain :: Symbol -> InfoTableBuilder m ()
   SetIdentArgsInfo :: Symbol -> [ArgumentInfo] -> InfoTableBuilder m ()
-  GetIdent :: Text -> InfoTableBuilder m (Maybe IdentKind)
+  GetIdent :: NameId -> InfoTableBuilder m (Maybe IdentKind)
```

Comment:
I need to discuss with Jan - he asked me to change from Text to NameId because he argued that we should only use NameId after scoping. This change is not related to Juvix REPL requirements.

Reply:
I've reverted the change to the infotable builder - I realised that Jan's comment was about something unrelated to this.